ee/ucp/kubernetes/storage/use-vsphere-volumes.md

```diff
@@ -0,0 +1,82 @@
+* `disk.EnableUUID` value on the worker VMs must be set to `True`.
+
+## Configure for Kubernetes
+
+Kubernetes cloud providers provide a method of provisioning cloud resources through Kubernetes via the `--cloud-provider` option. This is to ensure that the kubelet is aware that it must be initialized by ucp-kube-controller-manager before it is scheduled any work.
+
+```bash
+docker container run --rm -it --name ucp -e REGISTRY_USERNAME=$REGISTRY_USERNAME -e REGISTRY_PASSWORD=$REGISTRY_PASSWORD \
+  -v /var/run/docker.sock:/var/run/docker.sock \
+  "dockereng/ucp:3.1.0-tp2" \
+  install \
+  --host-address <HOST_ADDR> \
+  --admin-username admin \
+  --admin-password XXXXXXXX \
+  --cloud-provider=vsphere \
+  --image-version latest:
+```
+
+## Create a StorageClass
+
+1. Create a StorageClass with a user specified disk format.
+```bash
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: fast
+provisioner: kubernetes.io/vsphere-volume
+parameters:
+  diskformat: zeroedthick
+```
+For example, `diskformat` can be `thin`, `zeroedthick`, or `eagerzeroedthick`. The default format is `thin`.
+2. Create a StorageClass with a disk format on a user specified datastore.
+```bash
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: fast
+provisioner: kubernetes.io/vsphere-volume
+parameters:
+    diskformat: zeroedthick
+    datastore: VSANDatastore
+```
+You can also specify the `datastore` in the StorageClass. The volume will be created on the datastore specified in the storage class, which in this case is `VSANDatastore`. This field is optional. If the datastore is not specified, then the volume will be created on the datastore specified in the vSphere config file used to initialize the vSphere Cloud Provider.
+
+For more information on Kubernetes Storage Classes, see [Storage Classes](https://kubernetes.io/docs/concepts/storage/storage-classes/).
+
+## Deploy vSphere Volumes
+
+After you create a StorageClass, you can create [PersistentVolumes (PV)](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#introduction) that deploy volumes attached to hosts and mounted inside pods. A PersistentVolumeClaim (PVC) is a claim for storage resources that are bound to a PV when storage resources are granted. 
+
+We recommend that you use the StorageClass and PVC resources as these abstraction layers provide more portability as well as control over the storage layer across environments.
+
+To deploy vSphere volumes:
+
+1. [Create a PVC from the plugin](https://vmware.github.io/vsphere-storage-for-kubernetes/documentation/policy-based-mgmt.html). When you define a PVC to use the storage class, a PV is created and bound.
+2. [Refer to the PVC from the Pod](https://vmware.github.io/vsphere-storage-for-kubernetes/documentation/policy-based-mgmt.html).
```

Comment:
Should we state the action here? As in, what are we expecting them to do after they refer to the section on PVC from the pod? For example, should they create a pod?

Reply:
@anusha-ragunathan Proposing to change step 2 to "Create a reference to the PVC from the pod in the configuration file." Does that change the technical meaning?

Reply:
Should be okay. I'd leave out the "in the configuration file". In short, you can change it to "Create a reference to the PVC from the pod"